### PR TITLE
feat: add filter rules to setting page

### PR DESCRIPTION
**Pull request by hqwuzhaoyi**
Filter rules
- DOMAIN: Exact match; example.com should match example.com
- DOMAIN-SUFFIX: Suffix matching; example.com should match www.example.com
- DOMAIN-KEYWORD: Keyword matching; example should match www.example.com 
- Regular: expression matching; https?://mail.google.com/* should match https://mail.google.com/mail/u/0/#inbox

#60

### CODE REVIEW

File: src/components/FilterRules.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import { RuleType, FilterRuleItem } from "../types";
+
+const ruleTypes: { label: string; value: RuleType }[] = [
+  {
+    label: "DOMAIN",
```
Comment (MichaelYuhe) [inline src/components/FilterRules.tsx:5]: Label can be Domain, Domain suffix etc

File: src/options.tsx [hunk 2]
```diff
@@ -60,13 +70,23 @@ function BasicSettings() {
...
         <input
-          className="bg-gray-50 border w-64 border-gray-300 text-gray-900 text-sm rounded-lg 
+          className="bg-gray-50 border w-64 border-gray-300 text-gray-900 text-sm rounded-lg
               focus:ring-blue-500 focus:border-blue-500 block"
           value={apiURL}
           onChange={updateApiURL}
           id="api_url"
         />
       </div>
+      <div className="flex flex-col gap-y-2">
```
Comment (MichaelYuhe) [inline src/options.tsx:80]: Appreciate it if there can be an empty line break two parts of code

**Review by MichaelYuhe — CHANGES_REQUESTED** [review]

File: src/utils.ts [hunk 3]
```diff
@@ -35,3 +37,28 @@ export const DEFAULT_PROMPT: string =
   `Based on the URL: "{{tabURL}}" and title: "{{tabTitle}}", ` +
   `classify the browser tab type as one of the following: "{{types}}". ` +
   `Respond with only the classification keyword from the list.`;
+
+export function matchesRule(url: URL, rule: FilterRuleItem) {
+  const host = url.host;
+  const { type, rule: value } = rule;
+  if (!value) {
+    return false;
```
Comment (MichaelYuhe) [inline src/utils.ts:45]: maybe it's better move to the top of this function instead of get host first

File: src/options.tsx [hunk 4]
```diff
@@ -60,13 +70,23 @@ function BasicSettings() {
...
+        <label htmlFor="api_url" className="text-xl font-medium">
+          Filter Rule
```
Comment (MichaelYuhe) [inline src/options.tsx:82]: I think it's better to have a short description to explain what's filter rule for

File: src/components/FilterRules.tsx (hunk 1 above)
Comment (hqwuzhaoyi) [reply src/components/FilterRules.tsx:5]: thanks! done😁

File: src/options.tsx (hunk 2 above)
Comment (hqwuzhaoyi) [reply src/options.tsx:80]: thanks! done😁

File: src/options.tsx (hunk 4 above)
Comment (hqwuzhaoyi) [reply src/options.tsx:83]: CleanShot 2023-12-13 at 00 14 12@2x

Do you think my explanation will be clearer like this?

File: src/utils.ts (hunk 3 above)
Comment (hqwuzhaoyi) [reply src/utils.ts:44]: thanks! done😁

File: src/options.tsx (hunk 4 above)
Comment (MichaelYuhe) [reply src/options.tsx:83]: yes! it's much better

**Review by MichaelYuhe — APPROVED** [review]

**Pull request merged**